packages/kosu-genesis-cli/src/cli/main.ts

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
+import { Kosu } from "@kosu/kosu.js";
+import chalk from "chalk";
+import Web3 from "web3";
+
+import { generateGenesisFromBlock } from "../functions";
+
+import { cli, loadConsensusParameters, validateOptions } from ".";
+
+async function main(): Promise<void> {
+    validateOptions(cli.parse(process.argv));
+
+    const { chainId, providerUrl, snapshotBlock, startTime } = cli;
+
+    const consensusParameters = loadConsensusParameters(cli);
+
+    let kosu: Kosu;
+    try {
+        const provider = new Web3(providerUrl).currentProvider;
+        kosu = new Kosu({ provider });
+    } catch (error) {
+        throw new Error(`Failed to setup provider: ${error}`);
+    }
+
+    try {
+        const genesis = await generateGenesisFromBlock(
+            kosu,
+            chainId,
+            parseInt(snapshotBlock),
```

Comment:
if we pass (ie string) to `parseInt` it returns `NaN`, as `generateGenesisFromBlock` receives a `number` would that be interpreted as a zero?

Reply:
Not sure I fully understand the question, but what's happening is:
- `generateGenesisFromBlock` receives snapshot block as a `number`
- the snapshotBlock we get from the CLI is a string (but we do validation in cli/main.ts) so we `parseInt` it
- TypeScript also prevents us from passing a string (not if someone is using JS, of course) so there is that safety too